src/Microsoft.ML.Data/StaticPipe/StaticPipeExtensions.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Data;
+using System;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Core.Data;
+
+namespace Microsoft.ML.Data.StaticPipe
+{
+    public static class StaticPipeExtensions
+    {
+        /// <summary>
+        /// Asserts that a given data view has the indicated schema. If this method returns without
+        /// throwing then the view has been validated to have columns with the indicated names and types.
+        /// </summary>
+        /// <typeparam name="T">The type representing the view's schema shape</typeparam>
+        /// <param name="view">The view to assert the static schema on</param>
+        /// <param name="env">The host environment to keep in the statically typed variant</param>
+        /// <param name="outputDecl">The delegate through which we declare the schema, which ought to
+        /// use the input <see cref="SchemaAssertionContext"/> to declare a <see cref="ValueTuple"/>
+        /// of the <see cref="PipelineColumn"/> indices, properly named</param>
+        /// <returns>A statically typed wrapping of the input view</returns>
+        public static DataView<T> AssertStatic<[IsShape] T>(this IDataView view, IHostEnvironment env,
```

Comment:
I am not sure if we should have methods named/prefixed `Assert` but aren't in the same vein as `Debug.Assert`. I think it may be confusing.  How about `EnsureStatic`?